projects/components/src/table/controls/table-controls-api.ts

```diff
@@ -1,12 +1,56 @@
-import { KeyValue } from '@angular/common';
+import { Dictionary } from '@hypertrace/common';
 import { SelectOption } from '../../select/select-option';
+import { TableFilter } from '../table-api';
 
-export interface SelectFilter {
+export interface SelectControl {
   placeholder?: string;
-  options: SelectOption<KeyValue<string, unknown>>[];
+  default?: SelectOption<TableControlOption>;
+  options: SelectOption<TableControlOption>[];
 }
 
 export interface SelectChange {
-  select: SelectFilter;
-  value: KeyValue<string, unknown>;
+  select: SelectControl;
+  value: TableControlOption;
 }
+
+export interface CheckboxControl {
+  label: string;
+  value: boolean;
+  options: TableCheckboxOptions;
```

Comment:
should this be a trueOption and a falseOption both required - seems weird to ask for an array of two with known values in it. If you do leave it as the array, see comment below about type.

Reply:
I knew this might be a sticking point because it just feels wrong in many ways, but the reason it is an array to so that all controls can have the same data fetching shape.